oci/config/convert/runtime.go

```diff
@@ -123,31 +126,23 @@ func MutateRuntimeSpec(g rgen.Generator, rootfs string, image ispec.Image, manif
 
 	g.SetProcessUID(uint32(execUser.Uid))
 	g.SetProcessGID(uint32(execUser.Gid))
-	g.ClearProcessAdditionalGids()
+
 	for _, gid := range execUser.Sgids {
 		g.AddProcessAdditionalGid(uint32(gid))
 	}
 	if execUser.Home != "" {
 		g.AddProcessEnv("HOME", execUser.Home)
 	}
 
+	// Set optional fields
+	for exp := range image.Config.ExposedPorts {
+		g.AddAnnotation(exp, "")
+	}
```

Comment:
As above this needs to be done as per the spec. The key name and format is in the conversion doc:

> **org.opencontainers.image.exposedPorts** is the list of values that correspond to the [keys defined for `Config.ExposedPorts`](config.md) (string, comma-separated values).

Basically you need to set `org.opencontainers.image.exposedPorts` to the value of joining all of the keys in `image.Config.ExposedPorts` with commas.